src/astro/databases/postgres.py

```diff
@@ -26,3 +33,66 @@ def hook(self):
     def default_metadata(self) -> Metadata:
         # TODO: comment why
         return Metadata(database=self.hook.schema)
+
+    def load_pandas_dataframe_to_table(
+        self,
+        source_dataframe: pd.DataFrame,
+        target_table: Table,
+        if_exists: LoadExistStrategy = "replace",
+        chunk_size: int = DEFAULT_CHUNK_SIZE,
+    ) -> None:
+        """
+        Create a table with the dataframe's contents.
+        If the table already exists, append or replace the content, depending on the value of `if_exists`.
+
+        :param source_dataframe: Local or remote filepath
+        :param target_table: Table in which the file will be loaded
+        :param if_exists: Strategy to be used in case the target table already exists.
+        :param chunk_size: Specify the number of rows in each batch to be written at a time.
+        """
+
+        # To do - replace with create_schema() method.
+        conn = self.hook.get_connection(target_table.conn_id)
+        user = conn.login
+        schema = getattr(target_table.metadata, "schema", None)
+        create_schema_query = (
+            f"CREATE SCHEMA IF NOT EXISTS {schema} AUTHORIZATION {user}"
+        )
+        self.run_sql(sql_statement=create_schema_query, parameters={})
+
+        source_dataframe.to_sql(
+            target_table.name,
+            schema=schema.lower() if schema else None,
+            con=self.sqlalchemy_engine,
+            if_exists=if_exists,
+            chunksize=chunk_size,
+            method="multi",
+            index=False,
+        )
+
+    def get_table_qualified_name(self, table: Table) -> str:  # skipcq: PYL-R0201
+        """
+        Return table qualified name. This is Database-specific.
+        For instance, in Sqlite this is the table name. In Snowflake, however, it is the database, schema and table
+
+        :param table: The table we want to retrieve the qualified name for.
+        """
+        # Initially this method belonged to the Table class.
+        # However, in order to have an agnostic table class implementation,
+        # we are keeping all methods which vary depending on the database within the Database class.
+        if table.metadata and table.metadata.schema:
+            qualified_name = f"{table.metadata.schema.lower()}.{table.name}"
+        else:
+            qualified_name = table.name
+        return qualified_name
+
+    def table_exists(self, table: Table) -> bool:
+        """
+        Check if a table exists in the database.
+
+        :param table: Details of the table we want to check that exists
+        """
+        schema = getattr(table.metadata, "schema", None)
+        schema = schema.lower() if schema else None
```

Comment:
A comment might help on why we lowercase it here

Reply:
I'll add it